tests: Update submodules for bats

polarathene opened the pull request:
# Description

This PR updates `bats` submodules. Notably the `bats-assert` and `bats-support` submodules have since been migrated to the `bats-core` organization, while our current remote has been unmaintained since 2016.

- [`bats-core`](https://github.com/bats-core/bats-core) our last release of `1.4.1` (July 2021) updated to [`1.7.0` (May 2022)](https://github.com/bats-core/bats-core/releases/tag/v1.7.0). Some fixes and new `setup_suite()` + `teardown_suite()` options to run once before and after all individual test files (`setup_file()` / `teardown_file()`) are handled.
- [`bats-assert`](https://github.com/bats-core/bats-assert) has had a fair amount of updates. I've opted to update to the current master commit as no official release has been tagged since Nov 2018. There is a [request for a new tagged release](https://github.com/bats-core/bats-assert/issues/33) since Jan 2021. There is some new assertions since the last release, such as `assert_not_equal`.
- [`bats-support`](https://github.com/bats-core/bats-support) has had no new relevant activity for us since.

## Type of change

- [x] Improvement (non-breaking change that does improve existing functionality)

## Checklist:

- [ ] New and existing unit tests pass locally with my changes

martin-schulze-vireso commented:
Have a look at the reported error location (line 34) https://github.com/docker-mailserver/docker-mailserver/blob/2d0f5a0791bce4f87bba51993795f9eb393bb96e/test/setup-cli.bats#L32-L34

It clearly shows that there is no `run` which sets the `output` for `assert_success`. This is an error. The update of bats-assert pulled in code that now reports this error. The v2.0.0 did not have this check.

polarathene commented:
Argh, @martin-schulze-vireso just beat me to it! I had just looked into it myself and observed the same issue on our side :sweat_smile: 

https://github.com/docker-mailserver/docker-mailserver/blob/2d0f5a0791bce4f87bba51993795f9eb393bb96e/test/tests.bats#L641-L655

https://github.com/docker-mailserver/docker-mailserver/blob/2d0f5a0791bce4f87bba51993795f9eb393bb96e/test/setup-cli.bats#L32-L51

This is something I noticed recently when reviving the stale WIP encryption PR which had asserts that were actually using the last `run` unintentionally instead of using `run` on the actual command to be asserted.

polarathene commented:
That should fix the failures. Additional information regarding those changes is available in the commit messages.

georglauterbach commented:
I am currently re-running the tests as there were some weird quota errors. We'll see what happens in the second run.

polarathene commented:
Tests are passing now.

- The update revealed the mishandled `assert` calls.
- I removed the unnecessary `run` usage, which was consuming the errors silently.
- Fixed the tests :+1:
